catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/action/provider/RegistryPublicationActionProvider.java

```diff
@@ -0,0 +1,261 @@
+        }
+        List<Action> actions = new ArrayList<>();
+        List<String> currentPublications = publications.get(registryId);
+
+        for (RegistryStore registry : registryStores) {
+            //can't publish to yourself
+            if (!registry.isPushAllowed() || StringUtils.isBlank(registry.getRegistryId()) || registryId.equals(registry.getRegistryId())) {
+                continue;
+            }
+
+            if (currentPublications.contains(registry.getId())) {
+                actions.add(getAction(registryId, registry.getId(), false));
+            } else {
+                actions.add(getAction(registryId, registry.getId(), true));
+            }
+        }
+
+        return actions;
+    }
+
+    @Override
+    public String getId() {
+        return providerId;
+    }
+
+    @Override
+    public <T> boolean canHandle(T subject) {
+        return getRegistryId(subject) != null;
+    }
+
+    @Override
+    public void handleEvent(Event event) {
+        Metacard mcard = (Metacard) event.getProperty(METACARD_PROPERTY);
+        if (mcard == null || !mcard.getTags()
+                .contains(RegistryConstants.REGISTRY_TAG)) {
+            return;
+        }
+
+        String registryId = mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                .getValue()
+                .toString();
+        Attribute locationAttribute =
+                mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        List<String> locations = new ArrayList<>();
+        if (locationAttribute != null) {
+            locations = locationAttribute.getValues()
+                    .stream()
+                    .map(Object::toString)
+                    .collect(Collectors.toList());
+        }
+        if (event.getTopic()
+                .equals(CREATED_TOPIC) || event.getTopic()
+                .equals(UPDATED_TOPIC)) {
+            publications.put(registryId, locations);
+        } else if (event.getTopic()
+                .equals(DELETED_TOPIC)) {
+            publications.remove(registryId);
+        }
+    }
+
+    public void init() throws FederationAdminException {
+        List<Metacard> metacards = federationAdminService.getRegistryMetacards();
+        for (Metacard mcard : metacards) {
+            Attribute locations =
+                    mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+            if (locations != null) {
+                publications.put(mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                                .getValue()
+                                .toString(),
+                        locations.getValues()
+                                .stream()
+                                .map(Object::toString)
+                                .collect(Collectors.toList()));
+            } else {
+                publications.put(mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
+                        .getValue()
```

Comment:
Could also use this: https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#emptyList--